dask_expr/tests/test_resample.py

```diff
@@ -20,15 +20,56 @@ def df(pdf):
     yield from_pandas(pdf, npartitions=4)
 
 
-def test_resample_count(df, pdf):
-    result = df.resample("2T").count()
-    expected = pdf.resample("2T").count()
+@pytest.mark.parametrize(
+    "api",
+    [
+        "count",
+        "prod",
+        "mean",
+        "sum",
+        "min",
+        "max",
+        "first",
+        "last",
+        "var",
+        "std",
+        "size",
+        "nunique",
+        "median",
+        "quantile",
+        "ohlc",
+        "sem",
+    ],
+)
+def test_resample_apis(df, pdf, api):
+    result = getattr(df.resample("2T"), api)()
+    expected = getattr(pdf.resample("2T"), api)()
     assert_eq(result, expected)
 
-    result = df.resample("2T").count()["foo"]
-    expected = pdf.resample("2T").count()["foo"]
+    # No column output
+    if api not in ("size",):
+        result = getattr(df.resample("2T"), api)()["foo"]
+        expected = getattr(pdf.resample("2T"), api)()["foo"]
+        assert_eq(result, expected)
+
+        q = result.simplify()
+        eq = getattr(df["foo"].resample("2T"), api)().simplify()
+        assert q._name == eq._name
+
+
+def test_resample_agg(df, pdf):
+    def my_sum(vals, foo=None, *, bar=None):
+        return vals.sum()
+
+    # TODO: w/o .compute() assert_eq fails w/ dtype assertion error
+    result = df.resample("2T").agg(my_sum, "foo", bar="bar").compute()
+    expected = pdf.resample("2T").agg(my_sum, "foo", bar="bar")
+    assert_eq(result, expected)
+
+    result = df.resample("2T").agg(my_sum)["foo"].compute()
+    expected = pdf.resample("2T").agg(my_sum)["foo"]
     assert_eq(result, expected)
 
-    q = result.simplify()
-    eq = df["foo"].resample("2T").count().simplify()
+    q = df.resample("2T").agg(my_sum)["foo"].simplify()
```

Comment:
Can you block optimizations for agg in the beginning? I am not sure how agg works exactly, it could be that this is not allowed since the callable in agg can access bar and this optimising would raise

Reply:
Something like https://github.com/dask-contrib/dask-expr/pull/399/commits/403447aaecf1de033bf52c9612f19bc8f1f5f1f8?

Reply:
I'd prefer overriding ``_simplify_up`` in the Agg class

Reply:
Ya that makes sense. https://github.com/dask-contrib/dask-expr/pull/399/commits/63b327301047df3e6ee2583a892ee9b424ee6000